pkg/services/ngalert/state/state.go

```diff
@@ -119,12 +118,12 @@ func (a *State) SetError(err error, startsAt, endsAt time.Time) {
 }
 
 // SetNormal sets the state to Normal. It changes both the start and end time.
-func (a *State) SetNormal(reason string, startsAt, endsAt time.Time) {
+func (a *State) SetNormal(reason string, startsAt, endsAt time.Time, err error) {
 	a.State = eval.Normal
 	a.StateReason = reason
 	a.StartsAt = startsAt
 	a.EndsAt = endsAt
-	a.Error = nil
+	a.Error = err
```

Comment:
Do I understand correctly, that we set error here to provide this field to Alert that is sent to Alertamanger so user can get resolve notification with error in annotations?

Reply:
Didn't go that far. I set it to show the error here:

<img width="1019" alt="Captura de pantalla 2023-02-22 a las 22 41 38" src="https://user-images.githubusercontent.com/2112640/220765769-cb639405-7809-4935-a525-2627c12c29ca.png">